pando-rt/src/start.cpp

```diff
@@ -105,28 +120,43 @@ extern "C" int __start(int argc, char** argv) {
       // uniform distribution of potential target cores not including this scheduler core
       std::uniform_int_distribution<std::int8_t> dist(0, coreDims.x - 1);
 
+#ifdef PANDO_RT_USE_BACKEND_PREP
       std::vector<pando::Queue<pando::Task>::ProducerToken> ptoks;
       for(std::uint8_t i = 0; i < coreDims.x; i++){
-        const pando::Place dstPlace(thisPlace.node, thisPlace.pod,
-                                    pando::CoreIndex(i, 0));
-        auto* otherQueue =  pando::Cores::getTaskQueue(pando::Place{thisPlace.node, thisPlace.pod, pando::CoreIndex(i, 0)});
+        const pando::Place dstPlace(thisPlace.node, thisPlace.pod, pando::CoreIndex(i, 0));
+        auto* otherQueue =  pando::Cores::getTaskQueue(dstPlace);
         ptoks.push_back(otherQueue->makeProducerToken());
       }
+#endif
 
       do {
         // distribute tasks from the queue
-        if (auto task = queue->tryDequeue(ctok); task.has_value()) {
+#ifdef PANDO_RT_USE_BACKEND_PREP
+        auto task = queue->tryDequeue(ctok);
+#elif defined(PANDO_RT_USE_BACKEND_DRVX)
+        auto task = queue->tryDequeue();
+#endif
```

Comment:
these interfaces were designed to be the same, why has this been changed?

Reply:
Done